### PR TITLE
[EVM] Move generation of JUMPDEST to the AsmPrinter phase

### PR DESCRIPTION
**Pull request by vladimirradosavljevic**

### CODE REVIEW

File: llvm/lib/Target/EVM/EVMAsmPrinter.cpp [hunk 1]
```diff
@@ -105,8 +81,37 @@ void EVMAsmPrinter::emitInstruction(const MachineInstr *MI) {
...
                                 MF->getRegInfo());
 
-  unsigned Opc = MI->getOpcode();
-  if (Opc == EVM::DATASIZE_S || Opc == EVM::DATAOFFSET_S) {
+  switch (MI->getOpcode()) {
+  default:
+    break;
+  case EVM::PseudoCALL:
+  case EVM::PseudoRET: {
+    // TODO: Use PseudoInstExpansion and do this expansion in tblgen.
```
Comment (akiramenai) [inline llvm/lib/Target/EVM/EVMAsmPrinter.cpp]: Could you create a ticket for that?

File: llvm/lib/Target/EVM/EVMAsmPrinter.cpp [hunk 2]
```diff
@@ -105,8 +81,37 @@ void EVMAsmPrinter::emitInstruction(const MachineInstr *MI) {
...
+    Jump.setOpcode(EVM::JUMP_S);
+    EmitToStreamer(*OutStreamer, Jump);
+    return;
+  }
+  case EVM::PseudoJUMP:
+  case EVM::PseudoJUMPI: {
+    MCInst Push;
+    Push.setOpcode(EVM::PUSH4_S);
+
+    // TODO: Refactor EVMMCInstLower::Lower so we could use lowerOperand
```
Comment (akiramenai) [inline llvm/lib/Target/EVM/EVMAsmPrinter.cpp]: same as above

**Review by PavelKopyl — APPROVED** [review]
LGTM, thanks!

**Comment by akiramenai** [comment]
Could we create tickets for todos?

**Comment by vladimirradosavljevic** [comment]
> Could we create tickets for todos?

Done.

**Review by akiramenai — APPROVED** [review]
LGTM, please squash the TODO patch.

**Pull request merged**